workspace/exercise.go

```diff
@@ -59,3 +65,46 @@ func (e Exercise) HasMetadata() (bool, error) {
 	}
 	return false, err
 }
+
+// HasLegacyMetadata checks for the presence of a legacy exercise metadata file.
+// If there is no such file, it could also be an unrelated directory.
+func (e Exercise) HasLegacyMetadata() (bool, error) {
+	_, err := os.Lstat(e.LegacyMetadataFilepath())
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	if err == nil {
+		return true, nil
+	}
+	return false, err
+}
+
+// MigrateLegacyMetadataFile migrates a legacy metadata to the modern location.
+// This is a noop if the metadata file isn't legacy.
+// If both legacy and modern metadata files exist, the legacy file will be deleted.
+func (e Exercise) MigrateLegacyMetadataFile() (string, error) {
+	var str string
+	legacyMetadataFilepath := e.LegacyMetadataFilepath()
+	metadataFilepath := e.MetadataFilepath()
+
+	if _, err := os.Lstat(legacyMetadataFilepath); err != nil {
+		return "", nil
+	}
+	if err := createIgnoreSubdir(filepath.Dir(legacyMetadataFilepath)); err != nil {
+		return "", err
+	}
+	if _, err := os.Lstat(metadataFilepath); err != nil {
+		if err := os.Rename(legacyMetadataFilepath, metadataFilepath); err != nil {
+			return "", err
+		}
+		str = fmt.Sprintf("\nMigrated metadata to %s\n", metadataFilepath)
+		fmt.Fprintf(os.Stderr, str)
+	} else {
+		if err := os.Remove(legacyMetadataFilepath); err != nil {
+			return "", err
+		}
+		str = fmt.Sprintf("\nRemoved legacy metadata: %s\n", legacyMetadataFilepath)
+		fmt.Fprintf(os.Stderr, str)
+	}
+	return str, nil
```

Comment:
I think we can get rid of the `else` and move everything from the else down here.